docs/design/proposals/multi-version/multi-version.md

```diff
@@ -0,0 +1,259 @@
+management system known as the Control loop or Reconcilliating systems.
+
+Currently, ACK uses a code-genrator to generate everything that is need to
+build and run a controller: API definitions, CRDs, AWSResourceManager, helm
+charts etc... 98% of the controllers code is generated by the ACK community
+members. When a new service team introduces a new field, resource or
+operation all we do on the ACK side is bumping the aws-sdk-go version and
+regenerate the controller. :magic:
+
+However, when service teams/ack developers have to rename a field or make
+a field a kubernetes secret reference, they make significant changes to
+the CRD that are backward incompatible. In kubernetes world, CRD schemas
+should not be modified but rather introduced in a new CRD version.
+
+### Kubernetes and multi-version CRDs
+
+In a multi-version CRDs, each version of the resource is identified by a
+distinct version string (v1alpha1, v1beta2, v1 etc...). Kubernetes uses the
+version string to route API requests to the appropriate version of the
+resource.
+
+Even when dealing with multiple versions, we still need one controller handling
+the main "Hub" version, and use some "mutating" mechanisms to transform the
+non-main "Spoke" version to the hub version. This mechanism is known as the
+mutating webhooks.
+
+Mutating webhooks are a type of admission controller that allows you to modify
+incoming requests to Kubernetes API server before they are persisted. They can
+be used to modify or validate the content of a Kubernetes object before it's
+created or updated. There are three known type of admission controllers:
+- Mutating Webhooks
+- Defaulting Webhooks
+- Validating Webhooks
+
+Mutating webhooks work by intercepting incoming requests to the Kubernetes
+API server and forwarding them to a webhook endpoint. The webhook endpoint
+can then modify the request payload, add, remove or change fields or validate
+the request payload before it's persisted in etcd storage. Once the webhook
+endpoint modifies the request payload, the admission controller sends the
+updated payload to the API server for storage.
+
+In the context of multi-version CRDs, mutating webhooks can be used to
+automatically update older versions of custom resources to the latest version.
+The mutating webhook can modify the payload of an older version of the
+resource to match the new version of the resource. This can help you maintain
+backwards compatibility when you need to change the schema of a custom resource.
+
+[TODO Add schema explaining the admission webhooks + api-server + controller
+interactions]
+
+## Motifivation
+
+The purpose of this design document is to outline a plan for adding support
+for multi-versioning and generating mutating webhooks for ACK Controllers.
+Multi-versioning and mutating webhooks are important features that can help
+improve the reliability and maintainability of custom resources in Kubernetes.
```

Comment:
I would add another sentence explicitly stating something along the lines of "We think that supporting multi-version CRDs *for resources managed by ACK controllers* is important to users because ......."